src/vaadin-menu-bar-interactions-mixin.html

```diff
@@ -0,0 +1,346 @@
+          break;
+        case 'End':
+          increment = -1;
+          idx = buttons.length - 1;
+          break;
+        default:
+          // do nothing.
+      }
+
+      idx = this._getAvailableIndex(idx, increment, buttons);
+      if (idx >= 0) {
+        event.preventDefault();
+        const btn = buttons[idx];
+        const wasExpanded = currentBtn === this._expandedButton;
+        if (wasExpanded) {
+          this._close();
+        }
+        this._focusButton(btn);
+        if (wasExpanded && btn.item && btn.item.children) {
+          this.__openSubMenu(btn, {keepFocus: true});
+        }
+      }
+    }
+
+    _getAvailableIndex(index, increment, buttons) {
+      const totalItems = buttons.length;
+      let idx = index;
+      for (let i = 0; typeof idx === 'number' && i < totalItems; i++, idx += increment || 1) {
+        if (idx < 0) {
+          idx = totalItems - 1;
+        } else if (idx >= totalItems) {
+          idx = 0;
+        }
+
+        const btn = buttons[idx];
+        if (!btn.disabled && !btn.hasAttribute('hidden')) {
+          return idx;
+        }
+      }
+      return -1;
+    }
+
+    get _subMenu() {
+      return this.shadowRoot.querySelector('vaadin-context-menu');
+    }
+
+    __alignOverlayPosition(e) {
+      const overlay = e.target;
+      const getRect = () => this._expandedButton.getBoundingClientRect();
+      if (overlay.hasAttribute('bottom-aligned')) {
+        overlay.style.bottom = parseInt(getComputedStyle(overlay).bottom) + getRect().height + 'px';
+      }
+      if (overlay.hasAttribute('right-aligned')) {
+        overlay.style.right = parseInt(getComputedStyle(overlay).right) - getRect().width + 'px';
+      }
+    }
+
+    _itemsChanged(items, splices) {
+      const subMenu = this._subMenu;
+      if (subMenu && subMenu.opened) {
+        subMenu.close();
+      }
+    }
+
+    _onMouseOver(e) {
+      const button = Array.from(e.composedPath()).filter(el => el.item)[0];
+      if (button && button !== this._expandedButton && (this.openOnHover || (button.item.children && this._subMenu.opened))) {
+        this.__openSubMenu(button);
+      }
+    }
+
+    __onContextMenuKeydown(e) {
+      const item = e.target;
+      const list = item.parentNode;
+      if (e.keyCode === 38 && item === list.items[0]) {
+        this._close(true);
+      }
+      // ArrowLeft, or ArrowRight on non-parent submenu item
+      if (e.keyCode === 37 || (e.keyCode === 39 && !item._item.children)) {
+        // Prevent ArrowLeft from being handled in context-menu
+        e.stopImmediatePropagation();
+        this._navigateByKey(e);
+        const button = this.shadowRoot.activeElement;
+        if (button && button.item && button.item.children) {
+          this.__openSubMenu(button, {keepFocus: true});
+        }
+      }
+    }
+
+    __outsideClickListener(e) {
+      const path = e.composedPath();
+      if (path.indexOf(this) === -1 && path.indexOf(this._subMenu.$.overlay) === -1) {
```

Comment:
Should check if it targets any buttons, instead of checking if it targets the menu-bar.

Currently eg. in the demo, it is really easy to make an outside click without closing the overlay. The buttons consume only a portion of the menu-bar width, so the empty part of the menu-bar is easy to hit.

Reply:
Fixed.